### ISSUE TITLE
HPDCache is not compatible with WtDcacheWbufDepth > 2

### ISSUE THREAD

**Issue by cathales** [openhwgroup/cva6#2345]
Hello César,

Setting WtDcacheWbufDepth to 4 instead of 2 would improve superscalar `cv32a65x` performance by 3%.

However, it seems to be incompatible with HPDCache as this configuration change makes smoke-gen tests fail, while using WT cache it works.

**Comment by cfuguet** [comment]
> Hello César,
> 
> 
> 
> Setting WtDcacheWbufDepth to 4 instead of 2 would improve superscalar `cv32a65x` performance by 3%.
> 
> 
> 
> However, it seems to be incompatible with HPDCache as this configuration change makes smoke-gen tests fail, while using WT cache it works.

Hello @cathales,

There is no limitation on the number of entries (other than timing and area constraints) in the WB of the HPDcache so it should work.

How many bits do you have in the AXI interface ? You should have at least 3 for it to work: [000-011] are used for reads and writes and 111 is used for uncacheabke read and writes and AMOs

**Comment by cfuguet** [comment]
And additional question, are you using the veto-testharness environment or the UVM environment for the smoke tests ?

**Comment by cathales** [comment]
Good catch, UVM tests fail and smoke-gen runs using UVM. It might be a UVM issue and not a RTL issue.

@AyoubJalali what do you think about this?

**Comment by AyoubJalali** [comment]
@AEzzejjari what do you think ?

**Comment by JeanRochCoulon** [comment]
@AEzzejjari found and fix a bug in UVM #2379

@cathales you can try WtDcacheWbufDepth = 4 !

**Comment by cathales** [comment]
Thank to @AEzzejjari for the fix! However it would cost almost 7 kGates for an CoreMark/MHz gain less than 3% so I would not do perform the change.

**Comment by cathales** [comment]
Erratum: as this impacts only cache area, we will indeed increase this parameter.
